tests/Usages/NamespaceUsagesExcludeAttributeTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+declare(strict_types = 1);
+
+namespace Spaze\PHPStan\Rules\Disallowed\Usages;
+
+use PHPStan\Rules\Rule;
+use PHPStan\Testing\RuleTestCase;
+use Spaze\PHPStan\Rules\Disallowed\DisallowedNamespaceFactory;
+use Spaze\PHPStan\Rules\Disallowed\RuleErrors\DisallowedNamespaceRuleErrors;
+
+class NamespaceUsagesExcludeAttributeTest extends RuleTestCase
+{
+
+	protected function getRule(): Rule
+	{
+		$container = self::getContainer();
+		return new NamespaceUsages(
+			$container->getByType(DisallowedNamespaceRuleErrors::class),
+			$container->getByType(DisallowedNamespaceFactory::class),
+			$container->getByType(NamespaceUsageFactory::class),
+			[
+				[
+					'namespace' => 'PrivateModule\*',
+					'message' => 'no private modules',
+					'excludeWithAttribute' => [
+						'\Attributes\*Class',
+					],
+				],
+			]
+		);
+	}
+
+
+	public function testRule(): void
+	{
+		// Based on the configuration above, in this file:
```

Comment:
```suggestion
```

This is used to explain the test in the more general tests, let's drop it here, or you'd need to add the "expect this error on this line" lines too 😁